runtime/runtime/src/lib.rs

```diff
@@ -1483,6 +1484,217 @@ impl Runtime {
         state_update.commit(StateChangeCause::Migration);
     }
 
+    fn process_transaction<'a>(
```

Comment:
+1 to moving it to a new function, but can you do it in a separate PR? Can be before the main one. Otherwise it's hard to see the actual diff.

Reply:
Please at least split the refactor and moving into a method into two commits. To avoid funky interactions with signatures and such, maybe refactor first, and move to a method second.